pkgs/by-name/en/encore/tsparser.nix

```diff
@@ -0,0 +1,72 @@
+{
+  runtimesPath,
+  rustPlatform,
+  makeWrapper,
+  pkg-config,
+  protobuf,
+  openssl,
+  version,
+  src,
+  lib,
+  ...
+}: let
+  cargoLockFile = ./tsparser-Cargo.lock;
+in
+  rustPlatform.buildRustPackage {
+    pname = "tsparser-encore";
+
+    inherit src version;
+
+    nativeBuildInputs = [
+      makeWrapper
+      pkg-config
+      openssl
+    ];
+
+    buildInputs = [
+      protobuf
+    ];
+
+    buildAndTestSubdir = "tsparser";
+
+    doCheck = false;
+
+    patchPhase = ''
+      runHook prePatch
+
+      # the build complains about the diff between both cargo.lock files and
+      # is necessary to have a cargo.lock since it contains git deps
+      rm -rf tsparser/Cargo.lock
+      cp ${cargoLockFile} Cargo.lock
+
+      # to avoid to compile "broken" members
+      rm -rf runtimes
+      substituteInPlace Cargo.toml \
```

Comment:
why do you need this substitute? is that because of the stray lock file? from what I understood it should be enough to just set `buildAndTestSubdir` as you do on line 30 to build a sub-package

Reply:
If I remove those subtitute lines I have the following error:

![image](https://github.com/user-attachments/assets/73754f96-9e38-4d7a-ba76-91820a174a74)

I'm not a Rust expert, so if you have any ideas why that happens, I'm open.

Reply:
Did you also remove the "rm -rf runtimes"? I'm wondering if it just tries to load all the Cargo.toml files from the workspace maybe

Reply:
It was trying to compile every workspace member... Maybe I'm doing something wrong but that was a progress while trying to package this project.